src/Collection.php

```diff
@@ -28,13 +29,15 @@ public function __construct($input)
     {
         if (is_callable($input)) {
             $this->inputFactory = $input;
-            $this->input = $input();
-        } elseif (is_array($input)) {
+            $input = $input();
+        }
+
+        if (is_array($input)) {
             $this->input = new RecursiveArrayIterator($input);
```

Comment:
If the tests pass and the performance benchmarks won't get worse you can remove it. I'm not sure if it's there for a reason really.